src/nix/develop.cc

```diff
@@ -298,7 +298,7 @@ struct Common : InstallableCommand, MixProfile
         for (auto & var : savedVars)
             out << fmt("%s=\"$%s:$nix_saved_%s\"\n", var, var, var);
 
-        out << "export NIX_BUILD_TOP=\"$(mktemp -d -t nix-shell.XXXXXX)\"\n";
+        out << "export NIX_BUILD_TOP=\"$(mktemp -d -t nix-shell.XXXXXX)\"\nchmod 777 \"$NIX_BUILD_TOP\"\n";
```

Comment:
You are really sure about 777? I would expect something more like 755.

Reply:
`/tmp` on Linux is usually 777, that's what I went off of.

Reply:
Isn't the common perms for /tmp `1777` (note the sticky bit)? (At least it is on my systems.)

Reply:
You're right, I thought the sticky bit signified that it was a directory for whatever reason. Fixed.